exporter/src/test/java/io/zeebe/exporters/kafka/KafkaExporterIT.java

```diff
@@ -59,70 +62,95 @@ public void tearDown() {
     exporterConfiguration = null;
   }
 
-  @Test
-  public void shouldExportRecords() {
+  private void shouldExportRecordsUsingFormat(Format format) {
     // given
-    startZeebeBroker();
+    startZeebeBroker(format);
 
     // when
     exporterIntegrationRule.performSampleWorkload();
 
     // then
     exporterIntegrationRule.stop();
-    assertRecordsExported();
+    assertRecordsExported(format);
     exporterIntegrationRule = null;
   }
 
-  private void assertRecordsExported() {
-    final Map<RecordId, String> records = consumeAllExportedRecords();
-    exporterIntegrationRule.visitExportedRecords(r -> assertRecordExported(records, r));
+  @Test
+  public void shouldExportRecordsUsingJsonFormat() {
+    shouldExportRecordsUsingFormat(Format.JSON);
+  }
+
+  @Test
+  public void shouldExportRecordsUsingProtobufFormat() {
+    shouldExportRecordsUsingFormat(Format.PROTOBUF);
+  }
+
+  private void assertRecordsExported(Format format) {
+    final Map<RecordId, Object> records = consumeAllExportedRecords(format);
+    exporterIntegrationRule.visitExportedRecords(r -> assertRecordExported(format, records, r));
   }
 
   private void assertRecordExported(
-      final Map<RecordId, String> producedRecords, final Record<?> record) {
-    final String producedRecord =
+      final Format format, final Map<RecordId, Object> producedRecords, final Record<?> record) {
+    final Object producedRecord =
         producedRecords.get(new RecordId(record.getPartitionId(), record.getPosition()));
 
     assertThat(producedRecord).isNotNull();
-    JsonAssertions.assertThat(producedRecord).isJsonEqualTo(record.toJson());
+
+    switch (format) {
+      case JSON:
+        {
+          assertThat(producedRecord).isInstanceOf(String.class);
+          JsonAssertions.assertThat((String) producedRecord).isJsonEqualTo(record.toJson());
+          break;
+        }
+      case PROTOBUF:
+        {
+          // no need to verify the correctness of protocol buffers protocol, just verify its a valid
```

Comment:
Can you elaborate? What we mapped the record to the wrong message? Where would we catch if we forgot to map one record type in the serialize for example?

Reply:
This library should test all the code paths for the source code in this library, so as it currently stands - Yes we need a test for `ProtobufRecordDeserializer.unpackRecord`.  

Though since this library relies on the de/serialisation implemented in `zeebe-exporter-protobuf` the tests for the correctness of serialisation should be implemented there.

Infact what I propose is to move `ProtobufRecordDeserializer.unpackRecord` into `zeebe-exporter-protobuf` as `RecordTransformer.fromGenericRecord` and implement a test case for deserialization in there.  Then there will be a single place that does the packing/unpacking.

Reply:
I did this in https://github.com/zeebe-io/zeebe-exporter-protobuf/pull/67